src/lib/tls/tls13/tls_transcript_hash_13.cpp

```diff
@@ -58,15 +63,125 @@ Transcript_Hash_State Transcript_Hash_State::recreate_after_hello_retry_request(
    return ths;
    }
 
+namespace {
+
+// TODO: This is a massive code duplication of the client hello parsing code,
+//       as well as basic parsing of extensions. We should resolve this.
+//
+// Ad-hoc idea: When parsing the production objects, we could keep markers into
+//              the original buffer. E.g. the PSK extensions would keep its off-
+//              set into the entire client hello buffer. Using that offset we
+//              could quickly identify the offset of the binders list slice the
+//              buffer without re-parsing it.
+//
+// Finds the truncation offset in a serialization of Client Hello as defined in
+// RFC 8446 4.2.11.2 used for the calculation of PSK binder MACs.
```

Comment:
... yeah. The binder MACs are unfortunately a major cross-cutting concern through our layered implementation architecture. Probably it would be a good idea how other implementations are handling this. Maybe I missed a short cut. The current implementation works, but its sad to see this.